### PR TITLE
Added emission factors for Ontario

### LINKED ISSUE #4886: CA-ON (Ontario) Emission factors
**Is your feature request related to a problem? Please describe.**

What appear to be trustworthy emission factors have been computed for Ontario. See [here](https://github.com/electricitymaps/electricitymaps-contrib/discussions/4869) and [here](https://github.com/electricitymaps/electricitymaps-contrib/discussions/4791#discussioncomment-4295383) for discussions.

**Describe the solution you'd like**
Add the identified lifecycle emission factors for CA-ON

* Coal (not applicable, retired in 2014): 1,069gCO2/kWh
* Natural Gas: 497gCO2/kWh
* Hydro: 22gCO2/kWh
* Wind: 10.69gCO2/kWh
* Nuclear: 4.8gCO2/kWh
* Solar: TBD with INCERACV as was done for other zones, see [here](http://viewer.webservice-energy.org/incer-acv/app/)

Screenshot 2022-12-20 at 11 38 42

The source for those should be (except for solar)

`Mallia, Eric, and Geoffrey Lewis. "Life cycle greenhouse gas emissions of electricity generation in the province of Ontario, Canada." The International Journal of Life Cycle Assessment 18.2 (2013): 377-391.`

with a link to https://link.springer.com/article/10.1007/s11367-012-0501-0

### PR DESCRIPTION
**Pull request by Nil-Cipher**
## Issue
Closes #4886

## Description

Added emission numbers for various electricity generation types for Ontario.

### Double check

- [x] I have run `pnpx prettier --write .` and `poetry run format` to format my changes.

### CODE REVIEW

**Comment by Nil-Cipher** [comment]
@VIKTORVAV99 made PR for review!

**Review by VIKTORVAV99 — CHANGES_REQUESTED** [review]
Thanks for opening the PR!

Please note the formatting in the source object at the bottom and the source formatting in the specific values should match.

There should also not be any new lines in the sources.
(This feedback was also provided on the other open PR.)

**Comment by Nil-Cipher** [comment]
Copied the original source text.

**Comment by Nil-Cipher** [comment]
@VIKTORVAV99 should be good now.

**Comment by VIKTORVAV99** [comment]
I'll try and take a look later today. 👍🏼

**Comment by Nil-Cipher** [comment]
@VIKTORVAV99 updates?

**Comment by VIKTORVAV99** [comment]
Ah sorry @Nil-Cipher this slipped under my radar when doing all the dependabot dependency updates.
I'll take a look at it later today.

File: config/zones/CA-ON.yaml [hunk 1]
```diff
@@ -109,11 +113,18 @@ emissionFactors:
...
+      value: 22
     nuclear:
       datetime: '2013-01-01'
-      source: Mallia, E., Lewis, G. "Life cycle greenhouse gas emissions of electricity
-        generation in the province of Ontario, Canada." (2013)
+      source: "Mallia, E., and G. Lewis. “Life Cycle Greenhouse Gas Emissions of Electricity Generation in the Province of Ontario, Canada.” The International Journal of Life Cycle Assessment, vol. 18, no. 2, 2012, pp. 377–391, https://doi.org/10.1007/s11367-012-0501-0."
       value: 4.8
+    wind:
+      datetime: '2013-01-01'
+      source: "Mallia, E., and G. Lewis. “Life Cycle Greenhouse Gas Emissions of Electricity Generation in the Province of Ontario, Canada.” The International Journal of Life Cycle Assessment, vol. 18, no. 2, 2012, pp. 377–391, https://doi.org/10.1007/s11367-012-0501-0."
```
Comment (VIKTORVAV99) [inline config/zones/CA-ON.yaml:126]:
```suggestion
      source: Mallia, E., and G. Lewis. "Life Cycle Greenhouse Gas Emissions of Electricity Generation in the Province of Ontario, Canada." The International Journal of Life Cycle Assessment, vol. 18, no. 2, 2012, pp. 377–391
```

File: config/zones/CA-ON.yaml [hunk 2]
```diff
@@ -109,11 +113,18 @@ emissionFactors:
...
         value: 57.47323263502428
+    hydro:
+      datetime: '2013-01-01'
+      source: "Mallia, E., and G. Lewis. “Life Cycle Greenhouse Gas Emissions of Electricity Generation in the Province of Ontario, Canada.” The International Journal of Life Cycle Assessment, vol. 18, no. 2, 2012, pp. 377–391, https://doi.org/10.1007/s11367-012-0501-0."
+      value: 22
     nuclear:
       datetime: '2013-01-01'
-      source: Mallia, E., Lewis, G. "Life cycle greenhouse gas emissions of electricity
-        generation in the province of Ontario, Canada." (2013)
+      source: "Mallia, E., and G. Lewis. “Life Cycle Greenhouse Gas Emissions of Electricity Generation in the Province of Ontario, Canada.” The International Journal of Life Cycle Assessment, vol. 18, no. 2, 2012, pp. 377–391, https://doi.org/10.1007/s11367-012-0501-0."
```
Comment (VIKTORVAV99) [inline config/zones/CA-ON.yaml:122]:
```suggestion
      source: Mallia, E., and G. Lewis. "Life Cycle Greenhouse Gas Emissions of Electricity Generation in the Province of Ontario, Canada." The International Journal of Life Cycle Assessment, vol. 18, no. 2, 2012, pp. 377–391
```

File: config/zones/CA-ON.yaml [hunk 3]
```diff
@@ -109,11 +113,18 @@ emissionFactors:
       - datetime: '2021-01-01'
         source: Electricity Maps, 2021 average
         value: 57.47323263502428
+    hydro:
+      datetime: '2013-01-01'
+      source: "Mallia, E., and G. Lewis. “Life Cycle Greenhouse Gas Emissions of Electricity Generation in the Province of Ontario, Canada.” The International Journal of Life Cycle Assessment, vol. 18, no. 2, 2012, pp. 377–391, https://doi.org/10.1007/s11367-012-0501-0."
```
Comment (VIKTORVAV99) [inline config/zones/CA-ON.yaml:118]:
```suggestion
      source: Mallia, E., and G. Lewis. "Life Cycle Greenhouse Gas Emissions of Electricity Generation in the Province of Ontario, Canada." The International Journal of Life Cycle Assessment, vol. 18, no. 2, 2012, pp. 377–391
```

File: config/zones/CA-ON.yaml [hunk 4]
```diff
@@ -87,6 +87,10 @@ emissionFactors:
       - datetime: '2021-01-01'
         source: Electricity Maps, 2021 average
         value: 57.47323263502428
+    gas:
+      datetime: '2013-01-01'
+      source: "Mallia, E., and G. Lewis. “Life Cycle Greenhouse Gas Emissions of Electricity Generation in the Province of Ontario, Canada.” The International Journal of Life Cycle Assessment, vol. 18, no. 2, 2012, pp. 377–391, https://doi.org/10.1007/s11367-012-0501-0."
```
Comment (VIKTORVAV99) [inline config/zones/CA-ON.yaml:92]:
```suggestion
      source: Mallia, E., and G. Lewis. "Life Cycle Greenhouse Gas Emissions of Electricity Generation in the Province of Ontario, Canada." The International Journal of Life Cycle Assessment, vol. 18, no. 2, 2012, pp. 377–391
```

File: config/zones/CA-ON.yaml [hunk 5]
```diff
@@ -210,8 +221,6 @@ parsers:
   price: CA_ON.fetch_price
   production: CA_ON.fetch_production
 sources:
-  ? "Mallia, E., Lewis, G. \"Life cycle greenhouse gas emissions of electricity generation\
-    \ in the province of Ontario, Canada.\" Int J Life Cycle Assess 18, 377\u2013\
-    391 (2013)"
+  ? "Mallia, E., and G. Lewis. “Life Cycle Greenhouse Gas Emissions of Electricity Generation in the Province of Ontario, Canada.” The International Journal of Life Cycle Assessment, vol. 18, no. 2, 2012, pp. 377–391, https://doi.org/10.1007/s11367-012-0501-0."
```
Comment (VIKTORVAV99) [inline config/zones/CA-ON.yaml:224]:
```suggestion
  Mallia, E., and G. Lewis. "Life Cycle Greenhouse Gas Emissions of Electricity Generation in the Province of Ontario, Canada." The International Journal of Life Cycle Assessment, vol. 18, no. 2, 2012, pp. 377–391:
    link: https://doi.org/10.1007/s11367-012-0501-0
```
Don't forget to remove the duplicate link under this line.

**Comment by Nil-Cipher** [comment]
@VIKTORVAV99 Should be good now!

**Review by VIKTORVAV99 — COMMENTED** [review]
Why did you add back the weird double quotes and wrap it in single quotes again? This should not be needed for YAML.

(the doi link is also back in the values for the emission factors).

**Comment by Nil-Cipher** [comment]
> Why did you add back the weird double quotes and wrap it in single quotes again? This should not be needed for YAML.
> 
> (the doi link is also back in the values for the emission factors).

Hm, maybe I merged things in the wrong order, sorry about this.

**Comment by Nil-Cipher** [comment]
Okay, this should not have been this painful, but I think it is good.

**Comment by VIKTORVAV99** [comment]
Seems like I missed the last change in here and I'm sorry about that.

But this has been resolved in another PR now so closing this one.

(A tip for the future is to re-request a review after changes, it's hard to keep up with all the "normal" notifications from GitHub and review requests end up automatically in my tools so they are easier to keep track of.)

**Comment by madsnedergaard** [comment]
Seems like we forgot to actually close this after https://github.com/electricitymaps/electricitymaps-contrib/pull/5987 - thanks again for the work put into this, your work was carried over in the new PR and the final small things fixed :)

**Pull request closed without merging**